### ISSUE TITLE
Login with Google/Facebook not working as expected.

### ISSUE THREAD

**Issue by dhavalk-simform** [benwinding/react-admin-firebase#100]
To enable login with google and Facebook I got a reference from this [link](https://github.com/benwinding/react-admin-firebase/blob/master/src-demo/src/CustomLoginPage.js) written by @benwinding.

when the user clicks on Google / Facebook button to log in, its open popup to enter login details and popup closed and it's not redirecting to the home page and can see loading icon on the login screen.

@benwinding please look into this issue.
I've attached a gif to show you what is happening after login with Facebook.

when I see in the firebase user list new user is available in the list.

![ezgif com-crop](https://user-images.githubusercontent.com/62876889/78781756-6eed1600-79be-11ea-80f5-513a00fd3b2b.gif)

**Comment by dhstack** [comment]
Your title also says Google. Are you having a problem with that too?

I haven't tried Facebook, but Google authentication works for me.

**Comment by dhavalk-simform** [comment]
@dhstack yes, I've tried both FB and google and result is same for both the provider.

**Comment by dhstack** [comment]
What version of `react-admin` and `react-admin-firebase` are you using? Can also include the login page code you're using?

**Comment by trackwell-mike** [comment]
> What version of `react-admin` and `react-admin-firebase` are you using? Can also include the login page code you're using?

Thanks for your help! I work with @dhavalk-simform and wanted to get a quick answer to you. 

We're using `react-admin` 3.2.1 and `react-admin-firebase` 3.1.5.

@dhavalk-simform May have a more current branch he's working on that contains a more accurate version of the Login page code, but I'll include his most-recently committed version below:

```
import React from "react";
import { Login, LoginForm } from "react-admin";
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import firebase from "firebase";

// Configure FirebaseUI.
const uiConfig = {
  // Popup signin flow rather than redirect flow.
  signInFlow: "popup",
  // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function.
  signInSuccessUrl: "/User",
  // We will display Google and Facebook as auth providers.
  signInOptions: [
    firebase.auth.GoogleAuthProvider.PROVIDER_ID
    // firebase.auth.FacebookAuthProvider.PROVIDER_ID
  ]
};

const SignInScreen = () => (
  <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
);

const CustomLoginForm = props => (
  <div>
    <LoginForm {...props} />
    <SignInScreen />
  </div>
);

const CustomLoginPage = props => (
  <Login {...props}>
    <CustomLoginForm {...props} />
  </Login>
);

export default CustomLoginPage;
```

**Comment by dhstack** [comment]
@trackwell-mike The only thing I can think of is try changing `signInSuccessUrl: "/User",` to `signInSuccessUrl: "/",` and see if that works. Or making "Users" lowercase to "users".

Not sure what else may be going on. I'm using `react-admin 3.3.3` and `react-admin-firebase 3.1.6`. Here's the code that works for me, though it's a bit different than the demo. Hopefully that helps.

```
import React from 'react';
import { Login } from 'react-admin';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import firebase from '../firebase.js';
import Footer from '../components/Footer.js';

// Configure FirebaseUI
const uiConfig = {
  // Popup signin flow rather than redirect flow
  signInFlow: 'popup',
  // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function
  signInSuccessUrl: '/',
  // Auth providers to display
  signInOptions: [
    firebase.auth.GoogleAuthProvider.PROVIDER_ID,
    firebase.auth.EmailAuthProvider.PROVIDER_ID
  ],
  // Terms of service url
  tosUrl: '/terms',
  // Privacy policy url
  privacyPolicyUrl: '/privacy'
};

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  firebaseui: {
    padding: '15px'
  }
}));

const SignIn = props => {
  const classes = useStyles();

  return (
    <Login backgroundImage="" {...props}>
      <Container component="main" maxWidth="xs">
        <div className={classes.paper}>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          <StyledFirebaseAuth
            uiConfig={uiConfig}
            firebaseAuth={firebase.auth()}
            className={classes.firebaseui}
          />
        </div>
        <Footer />
      </Container>
    </Login>
  );
};

export default SignIn;
```

**Comment by 5achinJani** [comment]
facebook login is working. it was the unexpected logout that was the case here which is already fixed in #91 
This can be closed now.

**Comment by dhavalk-simform** [comment]
@5achinJani and @dhstack  thanks. I'm closing this issue.
